### ISSUE TITLE
SASS Compiler Breaks on "top-bottom" keyframe on from/to selector in keyframe animation. 

### ISSUE THREAD

**Issue by maxxcrawford** [Prinzhorn/skrollr-stylesheets#46]
When using **skrollr-styleshseets**, my SCSS compiler errors on the `bottom-top` selector, requesting an expec `0%`.  I'm using grunt/compass to compile the SCSS.

**Sample Code**

```
-skrollr-animation-name: bgParallax500;

@-skrollr-keyframes bgParallax500 {
    bottom-top {
        background-position: 50% -200px;
    }
    center {
        background-position: 50% 0px;
    }
    top-bottom {
        background-position: 50% 200px;
    }
}
```

I know the compiled CSS code works, but is there any way around the SCSS compile error?

**Comment by Prinzhorn** [comment]
that's invalid CSS (animation-name outside of a block). This compiles for me

``` css
#foo {
    -skrollr-animation-name: bgParallax500;
}

@-skrollr-keyframes bgParallax500 {
    bottom-top {
        background-position: 50% -200px;
    }
    center {
        background-position: 50% 0px;
    }
    top-bottom {
        background-position: 50% 200px;
    }
}
```

> sass --version
> Sass 3.2.14 (Media Mark)

**Comment by maxxcrawford** [comment]
Still getting the error. 

**Edit – I did try the non-compiled code, and it does work for me. (When linking to external skrollr.css file)**

#### SCSS Input Code

```
#foo {
    -skrollr-animation-name: bgParallax500;
}

@-skrollr-keyframes bgParallax500 {
    bottom-top {
        background-position: 50% -200px;
    }
    center {
        background-position: 50% 0px;
    }
    top-bottom {
        background-position: 50% 200px;
    }
}

```

#### Grunt/Compass Error

> error public/css/scss/styles.scss (Line 296 of public/css/scss/app/partials/_page_about.scss: Invalid CSS after "": expected keyframes selector (e.g. 10%), was "bottom-top")
> Compilation failed in 1 files.

# 

#### Compiler Software Versions
- Grunt 1.4.24
- Sass 3.4.0 (Selective Steve)
-  Compass 1.0.1 (Polaris)

**Comment by Prinzhorn** [comment]
You're right. I've updated to Sass 3.4 and it's broken now. I don't know if there's a solution and I also don't have time to search for one.

**Comment by maxxcrawford** [comment]
No worries. I just set up an additional stylesheet for skrollr to compensate! Thanks.

**Comment by teyepe** [comment]
Same problem here, since I've updated to 3.4...
This might be an answer/explaination: 
https://github.com/sass/sass/issues/1383
and https://github.com/sass/sass/issues/429 which refers to this:
https://github.com/Compass/compass/blob/master/core/stylesheets/compass/css3/_animation.scss#L37-L48

**Comment by OhmzTech** [comment]
Is this really a closed issue? Appears to be a problem still with 3.4.3, but came up with a potential fix if this is still an issue.

**Comment by Prinzhorn** [comment]
I don't think we can solve this. But I can keep it open if you want to.

**Comment by spyesx** [comment]
Arf... I'm having the same issues. I moved to Yosemite Beta to participate to their project so I had to update Node, Homebrew, Grunt, Ruby and finally... Compass. Now I can't compile any of my projects using skrollr-stylesheet. Looks like a bad news :/ If anyone has an idea about to fix it, feel free to share!

**Comment by teyepe** [comment]
Problem solved! Just upgrade to Sass 3.4.4: http://sass-lang.com/documentation/file.SASS_CHANGELOG.html

**Comment by teyepe** [comment]
<b>EDIT:</b><br>Problem partially fixed: You are allowed to use arbitary identifiers, e.g. 

<pre> 
@-skrollr-keyframes animation1 {
    #{center-top} {
        ...
    }
}</pre>

... but still you cannot use offset integers (e.g. <code>50-bottom</code> or <code>10p-center</code>)

**Comment by spyesx** [comment]
Just updated to the 3.4.5 and we still have the problem... I'm stuck on a project because of this and I can't go back to a previous version :(

**Comment by spyesx** [comment]
It looks like SASS won't solve the problem. Skrollr-stylesheet has to be updated to use valid CSS identifiers.
https://github.com/sass/sass/issues/1460#event-172981995

**Comment by Edgar-Saavedra** [comment]
Seems like sass -v 3.3.7 compiles despite the syntax errors.

**Comment by mzdr** [comment]
Hey guys! 

I ran into this issue as well because I was using selectors based on a percentage like `260p`, `5000p` and so on. Since the updates of Sass I wasn't able to compile those stylesheets anymore, throwing an error all the time like mentioned above. So I [forked](https://github.com/mzdr/skrollr-stylesheets) this repository and made some tiny adjustments so that you can write your `*.sass` files with percentage selectors and skrollr-stylesheet will detect them but use them as `*p` selectors.

For example:

``` sass
@-skrollr-keyframes example1
  0%
    top: 0%

  240%
    top: -100%

#example1
    -skrollr-animation-name: example1
```

In this case `#example1` will get `data-0p` and `data-240p` with its relevant properties.

Hope this helps anyone having this issue…

**Comment by Prinzhorn** [comment]
This has been fixed in 1.0.0 thanks to #47 . Prefix the numbers with `skrollr-` e.g.

``` css
@-skrollr-keyframes foo {
    skrollr-50p {
        top:10px;
    }
}
```
